flaskphiid/annotation.py

```diff
@@ -47,7 +54,20 @@ def from_annotations(anns):
         merged = MergedAnnotation()
         for ann in anns:
             merged.add_annotation(ann)
-        return merged
+        if merged.has_compatible_family_typelist:
+            return merged.split_annotations_by_subtypes()
+
+        return [merged]
+
+    @staticmethod
+    def from_unsplittable_annotations(anns):
```

Comment:
I'm prbly missing something here - but if we're just adding an annotation to the MergedAnnotation - do we need a separate function? it seems like an empty annotation wouldn't get this far based on previous checks?

Reply:
I think you're right-- initially this was in place to ensure we didn't re-enter the splitting function (since from_annotations is where we normally enter it to begin with)
However, when it _does_ re-enter, it exits out at the first check, so it's inefficient but correct logic.